File: app/src/main/java/com/walfud/contactsync_android/service/user/UserService.java
package com.walfud.contactsync_android.service.user;

import android.content.Context;
import android.text.TextUtils;

import com.walfud.contactsync_android.service.prefs.PrefsService;

/**
 * Created by walfud on 2017/4/20.
 */

public class UserService {

    private Context mContext;
    private PrefsService mPrefs;

    public UserService(Context context) {
        this.mContext = context;
        mPrefs = new PrefsService(context);
    }

    public boolean isLogin() {
        return !TextUtils.isEmpty(getToken());
    }

    public void changeUser(String id) {
        mPrefs.setOid(id);
    }

    public String getToken() {
        return mPrefs.getUserToken();
    }

    public void setToken(String token) {
        mPrefs.setUserToken(token);
    }
}

File: app/src/main/java/com/walfud/contactsync_android/service/user/UserService.kt
package com.walfud.contactsync_android.service.user

import android.text.TextUtils
import com.walfud.contactsync_android.service.prefs.PrefsService

/**
 * Created by walfud on 2017/4/20.
 */

object UserService {
    val isLogin: Boolean
        get() = !TextUtils.isEmpty(PrefsService.userTokenInternal)

    fun changeUser(id: String) {
        PrefsService.oid = id
    }

    var token: String
        get() = PrefsService.userToken
        set(token) {
            PrefsService.userToken = token
        }
}
